app/controllers/schedules_controller.rb

```diff
@@ -59,27 +60,36 @@ def edit
   end
 
   def create
+    # Check to see if it has a root task before creating a schedule
+    msg = check_root_task(schedule_params[:tasks_attributes])
+    if !msg.nil?
+      redirect_to new_schedule_path(schedule: params[:schedule]), notice: msg
```

Comment:
Use `flash` with the appropriate level instead of `notice`. You'll see examples in the various controllers' methods.